javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -202,7 +446,7 @@ final class Matcher private[regex] (
 
   // Seal the state
 
-  def toMatchResult(): MatchResult = new SealedResult(inputstr, lastMatch)
+  def toMatchResult(): MatchResult = new SealedResult(inputstr, lastMatch, startOf)
```

Comment:
I do not think this is correct. `startOf` depends on `ensureLastMath`, and therefore it must be replicated in `SealedResult`, which has its own `lastMatch`.

Reply:
Oh I see ! You mean we might do something like "continue to next match" on the matcher, but if at this point we already called startOf, it's evaluated and we cannot modify it anymore, right?
So even the code in the Matcher is wrong actually.

Reply:
Let's just replace the lazy val by a def in Matcher. Then it becomes correct.

Reply:
I did something more advanced so that it works fine. But now there is no caching if it is not a sealed match result. Should I add caching in the original class ?